fineract-provider/src/main/java/org/apache/fineract/infrastructure/entityaccess/service/FineractEntityAccessReadServiceImpl.java

```diff
@@ -70,23 +72,22 @@ public FineractEntityAccessReadServiceImpl(final PlatformSecurityContext context
      * This method returns the list of entity IDs as a comma separated list Or
      * null if there is no entity restrictions or if there
      */
-    @Override
-    public String getSQLQueryInClause_WithListOfIDsForEntityAccess(Long firstEntityId, FineractEntityType firstEntityType,
-            FineractEntityAccessType accessType, FineractEntityType secondEntityType, boolean includeAllOffices) {
-        Collection<FineractEntityAccessData> accesslist = retrieveEntityAccessFor(firstEntityId, firstEntityType, accessType, secondEntityType,
-                includeAllOffices);
+    @SuppressWarnings("null")
+	@Override
+    public String getSQLQueryInClause_WithListOfIDsForEntityAccess( FineractEntityType firstEntityType,
+           final Long relId, final Long fromEntityId, boolean includeAllOffices) {
+        Collection<FineractEntityToEntityMappingData> accesslist = retrieveEntityAccessFor(firstEntityType, relId, fromEntityId, 
+        		includeAllOffices);
         String returnIdListStr = null;
         StringBuffer accessListCSVStrBuf = null;
         if ((accesslist != null) && (accesslist.size() > 0)) {
-            logger.debug("Found " + accesslist.size() + " access type restrictions while getting entity access configuration for "
-                    + firstEntityType.getType() + ":" + firstEntityId + " with type " + accessType.toStr() + " against "
-                    + secondEntityType.getType());
+           
             accessListCSVStrBuf = new StringBuffer(" ");
-            for (int i = 0; i < accesslist.size(); i++) {
-                FineractEntityAccessData accessData = (FineractEntityAccessData) accesslist.toArray()[i];
-                if (accessData == null) { throw new FineractEntityAccessConfigurationException(firstEntityId, firstEntityType, accessType,
-                        secondEntityType); }
-                if (accessData.getSecondEntity().getId() == 0) { // If there is
+          
+          for (int i = 0; i < accesslist.size(); i++) {
+            	FineractEntityToEntityMappingData accessData = (FineractEntityToEntityMappingData) accesslist.toArray()[i];
+                if (accessData == null) {  }
```

Comment:
if accessData is null what we have to do?
next line you are accessing the accessData, which might throw null pointer exception.